OData/src/System.Web.OData/OData/Routing/Conventions/AttributeRoutingConvention.cs

```diff
@@ -205,15 +205,17 @@ public string SelectController(ODataPath odataPath, HttpRequestMessage request)
                 {
                     IHttpActionSelector actionSelector = controller.Configuration.Services.GetActionSelector();
                     ILookup<string, HttpActionDescriptor> actionMapping = actionSelector.GetActionMapping(controller);
-                    IEnumerable<HttpActionDescriptor> actions = actionMapping.SelectMany(a => a);
-                    string prefix = GetODataRoutePrefix(controller);
-
-                    foreach (HttpActionDescriptor action in actions)
+                    HttpActionDescriptor[] actions = actionMapping.SelectMany(a => a).ToArray();
+ 
+                    foreach (string prefix in AttributeRoutingConvention.GetODataRoutePrefixes(controller))
                     {
-                        IEnumerable<ODataPathTemplate> pathTemplates = GetODataPathTemplates(prefix, action);
-                        foreach (ODataPathTemplate pathTemplate in pathTemplates)
+                        foreach (HttpActionDescriptor action in actions)
                         {
-                            attributeMappings.Add(pathTemplate, action);
+                            IEnumerable<ODataPathTemplate> pathTemplates = GetODataPathTemplates(prefix, action);
+                            foreach (ODataPathTemplate pathTemplate in pathTemplates)
+                            {
+                                attributeMappings.Add(pathTemplate, action);
```

Comment:
Add logic to handle duplicated route prefixes:
[ODataRoutePrefix("VipCustomer")]
[ODataRoutePrefix("VipCustomer")]